pipeline/gene_to_nissl.py

```diff
@@ -0,0 +1,218 @@
+DATA_FOLDER = pathlib.Path(__file__).resolve().parent.parent / "data"
+
+
+def parse_args():
+    """Parse arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--gene-path",
+        type=Path,
+        help="""\
+        Path to Gene Expression.
+        """,
+    )
+    parser.add_argument(
+        "--metadata-path",
+        type=Path,
+        help="""\
+        Path to json containing metadata of gene expression.
+        """,
+    )
+    parser.add_argument(
+        "--nissl-path",
+        type=Path,
+        default=DATA_FOLDER / "ara_nissl_25.nrrd",
+        help="""\
+        Path to Nissl Volume.
+        """,
+    )
+    parser.add_argument(
+        "--output-dir",
+        type=Path,
+        help="""\
+        Path to directory where to save the results.
+        """,
+    )
+    return parser.parse_args()
+
+
+def check_and_load(volume_path: pathlib.Path | str, normalize: bool = True) -> np.array:
+    """Load volume if path exists.
+
+    Parameters
+    ----------
+    volume_path
+        File path.
+    normalize
+        If True, normalize the image between 0 and 1.
+        Otherwise, keep the raw values.
+
+    Returns
+    -------
+    volume : np.array
+        Loaded volume.
+    """
+    volume_path = Path(volume_path)
+    if not volume_path.exists():
+        raise ValueError(f"The specified path {volume_path} does not exist.")
+
+    if volume_path.suffix == ".nrrd":
+        img, header = nrrd.read(volume_path)
+
+    elif volume_path.suffix == ".npy":
+        img = np.load(volume_path)
+
+    else:
+        raise ValueError(f"The extension {volume_path.suffix} is not supported.")
+
+    img = img.astype(np.float32)
+    if normalize:
+        img = (img - img.min()) / (img.max() - img.min())
+
+    return img
+
+
+def registration(
+    nissl_volume: np.array, gene_volume: np.array, section_numbers: np.array
+) -> np.array:
+    """Compute registration transform between a couple of volumes.
+
+    Parameters
+    ----------
+    nissl_volume
+        Nissl volume (fixed volume during registration).
+    gene_volume
+        Gene to register.(moving volume during registration).
+    section_numbers
+        Section numbers of every gene slice in gene volume.
+
+    Returns
+    -------
+    warped_genes : np.array
+        Warped slice.
+    """
+    rgb = False
+    if gene_volume.ndim == 4:
+        rgb = True
+
+    warped_genes = []
+    for i, (section_number, gene_slice) in enumerate(zip(section_numbers, gene_volume)):
+        try:
+            nissl_slice = nissl_volume[int(section_number)]
+        except IndexError:
+            continue
+
+        if rgb:
+            gene_slice_rgb = gene_slice.copy()
+            gene_slice = rgb2gray(gene_slice)
+
+        nii_data = register(nissl_slice, gene_slice, is_atlas=False)
+
+        if rgb:
+            warped = np.zeros_like(gene_slice_rgb)
+            warped[:, :, 0] = transform(gene_slice_rgb[:, :, 0], nii_data)
```

Comment:
I would imagine that `antspy` warping RGB images. I guess it it is your helper `transform` that does not really support it, right?

Reply:
We tried at some point and it is not working. If I remember correctly, it was mainly because one image is RGB and the other one is grayscale.